include/dev/RedundantADC.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <EVT/io/ADC.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace RedundantADC {
+
+/**
+ * This class allows processing readings from multiple ADCs and checking for redundancy.
+ */
+class RedundantADC {
+public:
+    /**
+     * @brief Enum class representing the status of RedundantADC processing.
+     */
+    enum class Status {
+        OK = 0,                /**< No error. */
+        OFF_BY_ONE_ERROR = 1,  /**< The reading is off by 1*/
+        MARGIN_ERROR = 2,      /**< The reading is off by less than the specific margin*/
+        COMPARISON_ERROR = 3,  /**< The readings do not match.*/
+        CAN_LOOPBACK_ERROR = 4 /**< CAN loopback data does not match.*/
+    };
+
+    /**
+     * @brief Constructs a RedundantADC object with the given ADC instances.
+     *
+     * @param[in] adc0 The first ADC instance.
+     * @param[in] adc1 The second ADC instance.
+     * @param[in] adc2 The third ADC instance.
+     */
+    RedundantADC(IO::ADC& adc0, IO::ADC& adc1, IO::ADC& adc2);
+
+    /**
+     * @brief Process readings from the ADCs and check for redundancy.
+     *
+     * This function reads values from three ADCs and checks for redundancy.
+     *
+     * @param[in] val1 Reference to the variable to store the value read from the first ADC.
+     * @param[in] val2 Reference to the variable to store the value read from the second ADC.
+     * @param[in] val3 Reference to the variable to store the value read from the third ADC.
+     * @return RedundantADC::Status The status of the processing.
+     */
+    RedundantADC::Status process(uint32_t& val1, uint32_t& val2, uint32_t& val3);
```

Comment:
This function should only return one value, the consensus value of the ADCs. If there is a major error, set this value to 0. It is the calling code's responsibility to read and handle the error but not to figure out which throttle reading to use

```suggestion
    RedundantADC::Status process(uint32_t& counts);
```

Also, we use `process` for our board functions, but a name like `readVoltage` would be more sensible for this function

Reply:
This function still needs to be modified to return one voltage instead of 3

Reply:
how would i handle off by one and margin errors by only returning 1 voltage?